### ISSUE TITLE
If Neo4j is hit hard it crashes - probably too many parallel (or left open) HTTP requests - investigate

### ISSUE THREAD

**Issue by GearsAD** [GearsAD/CloudGraphs.jl#17]

**Comment by dehann** [comment]
@pvazteixeira is seeing this via Caesar.jl also...

**Comment by dehann** [comment]
also see #15

**Comment by pvazteixeira** [comment]
Maybe this is related: I was just checking neo4j's log (on my machine) for any indication of what might be wrong and noticed it is chock full of dropped transactions (many per second), e.g.:

```
2017-04-28 21:24:17.605+0000 INFO  Transaction with id 23305 has been automatically rolled back.
```

The transaction ids are sequential, and these it gets above 20+ of these in 1ms.

**Comment by dehann** [comment]
also see dehann/IncrementalInference.jl#9

**Comment by dehann** [comment]
xref glesica/Neo4j.jl#12

**Comment by GearsAD** [comment]
Yep we spoke to Neo4j maintainers - I've worked in commits for WebAPI. We
may have to refactor the CloudGraphsIntegration code in Caesar/RoME.

Can we discuss tomorrow?

**Comment by dehann** [comment]
This issue was fixed with `commit(tx)` after read only transactions as in the Neo4j issue 12 mentioned about, right?

My suggestion for CloudGraphIntegration is as follows:
Do all graph related operations in `DistributedFactorGraphs.jl` (which could depend on either Synchrony or CloudGraphs).  This maximizes symmetry between Client and Server.  Then only use CloudGraphs.jl on server side.  

Ultimately the Caesar install would only pull down Synchrony and not CloudGraphs.  We should move `CloudGraphIntegration` out of Caesar and somehow into the server version of DistributedFG.  I quite like the callback strategy currently used:  That means that the `CloudGraphIntegration` functionality can live downstream of DistributedFactorGraphs.
